client-app/shared/cart/components/add-to-cart.vue

```diff
@@ -101,66 +97,84 @@ function onInput(value: number): void {
 async function onChange() {
   loading.value = true;
 
-  let lineItem = getLineItem(cart.value?.items);
+  try {
+    const lineItem = getLineItem(cart.value?.items);
+    const mode = lineItem ? AddToCartModeType.Update : AddToCartModeType.Add;
+    const updatedCart = await updateOrAddToCart(lineItem, mode);
 
-  let updatedCart: ShortCartFragment | undefined;
+    if (isConfigurable.value && mode === AddToCartModeType.Add) {
+      loading.value = false;
+      return;
+    }
 
-  const isAlreadyExistsInTheCart = !!lineItem;
-  const mode = isAlreadyExistsInTheCart && !isConfigurable.value ? AddToCartModeType.Update : AddToCartModeType.Add;
+    const updatedLineItem = getLineItem(updatedCart?.items);
+    handleUpdateResult(updatedLineItem, mode);
+  } finally {
+    loading.value = false;
+  }
+}
 
-  if (mode === AddToCartModeType.Update) {
-    updatedCart = await changeItemQuantity(lineItem!.id, enteredQuantity.value || 0);
-  } else {
-    const inputQuantity = enteredQuantity.value || minQty.value;
-    const configurationSections = isConfigurable.value ? selectedConfigurationInput.value : undefined;
-    updatedCart = await addToCart(product.value.id, inputQuantity, configurationSections);
-
-    /**
-     * Send Google Analytics event for an item added to cart.
-     */
-    ga.addItemToCart(product.value, inputQuantity);
-    void pushHistoricalEvent({
-      eventType: "addToCart",
-      sessionId: cart.value?.id,
-      productId: product.value.id,
-      storeId: globals.storeId,
-    });
+async function updateOrAddToCart(lineItem: ShortLineItemFragment | undefined, mode: AddToCartModeType) {
+  if (mode === AddToCartModeType.Update && !!lineItem) {
+    return isConfigurable.value
+      ? await changeCartConfiguredItem(lineItem.id, enteredQuantity.value || 0, selectedConfigurationInput.value)
+      : await changeItemQuantity(lineItem.id, enteredQuantity.value || 0);
```

Comment:
you sure needed call changeItemQuantity if enteredQuantity.value not positive?

Reply:
No needed indeed, just kept the same logic as it was. Now changed - if there is no enteredQuantity do nothing and return current short cart